From 2cd57e968f57537c01811dc244f12facf42ad8a3 Mon Sep 17 00:00:00 2001
From: Ronald Claveau <linux-kernel-dev@aliel.fr>
Date: Wed, 22 Apr 2026 11:42:41 +0200
Subject: [PATCH 89/91] arm64: dts: amlogic: t7: Add PCIE PHY node

The T7 integrates a PCIe PHY derived from the G12A USB3/PCIe combo PHY.

Signed-off-by: Ronald Claveau <linux-kernel-dev@aliel.fr>
Signed-off-by: Nick Xie <nick@khadas.com>
Change-Id: I9dcfc79a35466ddc9469587353183e85b3179a27
---
 arch/arm64/boot/dts/amlogic/amlogic-t7.dtsi | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/arch/arm64/boot/dts/amlogic/amlogic-t7.dtsi b/arch/arm64/boot/dts/amlogic/amlogic-t7.dtsi
index 58e744dbe4f8..0a724d0ce5b7 100644
--- a/arch/arm64/boot/dts/amlogic/amlogic-t7.dtsi
+++ b/arch/arm64/boot/dts/amlogic/amlogic-t7.dtsi
@@ -975,6 +975,18 @@ a53_tsensor: temperature-sensor@22000 {
 				amlogic,secure-monitor = <&sm 2>;
 			};
 
+			pcie_phy: phy@2a000 {
+				compatible = "amlogic,t7-pcie-phy", "amlogic,g12a-usb3-pcie-phy";
+				reg = <0x0 0x2a000 0x0 0x2000>;
+				clocks = <&pcie CLKID_PCIE_PLL>;
+				clock-names = "ref_clk";
+				resets = <&reset RESET_PCIE_PHY>;
+				reset-names = "phy";
+				assigned-clocks = <&pcie CLKID_PCIE_PLL>;
+				assigned-clock-rates = <100000000>;
+				#phy-cells = <1>;
+			};
+
 			pwm_ao_ef: pwm@30000 {
 				compatible = "amlogic,t7-pwm", "amlogic,meson-s4-pwm";
 				reg = <0x0 0x30000 0x0 0x24>;
-- 
2.53.0

